values.tf

```diff
@@ -1,13 +1,162 @@
 locals {
   values_default = yamlencode({
-    # add default values here
+    "config" : {
+      "localDnsIp" : "169.254.20.11",
+      "zones" : {
+        ".:53" : {
+          "plugins" : {
+            "errors" : true,
+            "reload" : true,
+            "debug" : false,
+            "log" : {
+              "format" : "combined",
+              "classes" : "all"
+            },
+            "cache" : {
+              "parameters" : 30,
+              "denial" : {
+                size : 0,
+                ttl : 1
+              },
+              "success" : {},
+              "prefetch" : {},
+              "serve_stale" : false
+            },
+            "forward" : {
+              "parameters" : "172.20.0.10",
+              "force_tcp" : true,
+              "prefer_udp" : false,
+              "policy" : "",
+              "max_fails" : "",
+              "expire" : "",
+              "health_check" : "",
+              "except" : ""
+            },
+            "prometheus" : true,
+            "health" : {
+              "port" : 8080
+            }
+          }
+        },
+        "ip6.arpa:53" : {
+          "plugins" : {
+            "errors" : true,
+            "reload" : true,
+            "debug" : false,
+            "log" : {
+              "classes" : "error"
+            },
+            "cache" : {
+              "parameters" : 30
+            },
+            "forward" : {
+              "parameters" : "172.20.0.10",
+              "force_tcp" : true
+            },
+            "prometheus" : true,
+            "health" : {
+              "port" : 8081
+            }
+          }
+        },
+        "in-addr.arpa:53" : {
+          "plugins" : {
+            "errors" : true,
+            "reload" : true,
+            "debug" : false,
+            "log" : {
+              "classes" : "error"
+            },
+            "cache" : {
+              "parameters" : 30
+            },
+            "forward" : {
+              "parameters" : "172.20.0.10",
+              "force_tcp" : true
+            },
+            "prometheus" : true,
+            "health" : {
+              "port" : 8082
+            }
+          }
+        }
+      }
+    },
+    "useHostNetwork" : true,
+    "priorityClassName" : "system-node-critical",
+    "podAnnotations" : {},
+    "podSecurityContext" : {},
+    "securityContext" : {
+      "privileged" : true
+    },
+    "readinessProbe" : null,
+    "serviceAccount" : {
+      "create" : true,
+      "annotations" : {
+        "eks.amazonaws.com/role-arn" : "arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${aws_iam_role.this[0].name}"
+      },
+      "name" : var.service_account_name
```

Comment:
Can we interpolate the service name here by using random_pet instead of doing it in every installation method ?

```
    "serviceAccount" : {
        "name" : "${var.service_account_name}-${one(random_pet.release_name_suffix[*].id)}"
    }
```

Reply:
This will cause a cycle 
```
Cycle: module.eks-common.module.node-local-dns.local.values_default (expand), module.eks-common.module.node-local-dns.random_pet.release_name_suffix, module.eks-common.module.node-local-dns.local.helm_release_name (expand)
```
`argo` and `argo-helm` methods do not change the service account name anyway. Only for `helm` deployment method i had to patch the values in later part for the terraform and rename the sa name for each new release.